orca-queue/src/test/kotlin/com/netflix/spinnaker/orca/q/TrafficShapingQueueSpec.kt

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2017 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.netflix.spinnaker.orca.q
+
+import com.netflix.spectator.api.NoopRegistry
+import com.netflix.spectator.api.Registry
+import com.netflix.spinnaker.orca.pipeline.model.Pipeline
+import com.netflix.spinnaker.orca.q.memory.InMemoryQueue
+import com.nhaarman.mockito_kotlin.*
+import org.jetbrains.spek.api.dsl.describe
+import org.jetbrains.spek.api.dsl.it
+import org.jetbrains.spek.api.lifecycle.CachingMode
+import org.jetbrains.spek.subject.SubjectSpek
+import java.time.Clock
+
+object TrafficShapingQueueSpec : SubjectSpek<TrafficShapingQueue>({
+  val queueImplCallback: QueueCallback = mock()
+  val queueImpl: Queue = InMemoryQueue(Clock.systemDefaultZone(), deadMessageHandler = mock(), publisher = mock())
+  val registry: Registry = NoopRegistry()
+  val interceptor: TrafficShapingInterceptor = mock()
+
+  subject(CachingMode.GROUP) {
+    whenever(interceptor.supports(any())) doReturn true
+    TrafficShapingQueue(queueImpl, registry, listOf(interceptor))
+  }
+
+  beforeGroup {
+    reset(queueImplCallback)
+    subject.push(StartStage(Pipeline::class.java, "1", "foo", "1"))
+  }
+
+  describe("when no interceptors are triggered") {
+    whenever(interceptor.interceptPoll()) doReturn false
+    whenever(interceptor.interceptMessage(any())).doReturn(null as TrafficShapingInterceptorCallback?)
```

Comment:
this needs to be done inside a `beforeGroup`. Otherwise it's evaluated when Spek runs through and builds the tree—before any of the tests start running. It's okay to set constants up like that but not something with mutable state like a mock.